src/darwin/Framework/CHIP/MTRDevice_Internal.h

```diff
@@ -120,8 +120,19 @@ MTR_DIRECT_MEMBERS
     // Our controller.  Declared nullable because our property is, though in
     // practice it does not look like we ever set it to nil.
     MTRDeviceController * _Nullable _deviceController;
+
+    // Whether this device has been accessed via the public deviceWithNodeID API
+    // (as opposed to just via the internal _deviceWithNodeID).
+    BOOL _accessedViaPublicAPI;
```

Comment:
Perhaps this should be renamed to: _readThoughsAllowed to match what it is trying to guard against: Allow it to be used without requiring readThroughs

Reply:
I guess that depends on whether we will have any other behavior branching on this state in the future.

And more importantly, _readThoughsAllowed doesn't make sense on the base MTRDevice, since the control over the read-through behavior it has only applies to MTRDevice_Concrete.  It doesn't affect MTRDevice_XPC behavior at all.

Reply:
Sounds good